### PR TITLE
fix: API types enum validations

### PR DESCRIPTION
**Pull request by mateusoliveira43**
# Please add a summary of your change

Fix lack of enum valiadtions for API types, in `pkg/apis/velero` folder, and clean up around constants.

# Does your change fix a particular issue?

No.

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: pkg/apis/velero/v1/backup_repository_types.go [hunk 1]
```diff
@@ -51,9 +51,6 @@ const (
 	BackupRepositoryPhaseNew      BackupRepositoryPhase = "New"
 	BackupRepositoryPhaseReady    BackupRepositoryPhase = "Ready"
 	BackupRepositoryPhaseNotReady BackupRepositoryPhase = "NotReady"
-
-	BackupRepositoryTypeRestic string = "restic"
```
Comment (reasonerjt) [inline pkg/apis/velero/v1/backup_repository_types.go:55]: I don't have a very strong opinion but why is this needed?

File: config/crd/v2alpha1/bases/velero.io_datauploads.yaml [hunk 2]
```diff
@@ -115,6 +115,8 @@ spec:
               snapshotType:
                 description: SnapshotType is the type of the snapshot to be backed
                   up.
+                enum:
+                - CSI
```
Comment (reasonerjt) [inline config/crd/v2alpha1/bases/velero.io_datauploads.yaml:119]: Not sure if we really want this 
cc @Lyndon-Li

File: pkg/apis/velero/v1/backup_repository_types.go (hunk 1 above)
Comment (mateusoliveira43) [reply pkg/apis/velero/v1/backup_repository_types.go:55]: If you mean the variable, it is used in these places https://github.com/search?q=repo%3Avmware-tanzu%2Fvelero%20BackupRepositoryTypeRestic&type=code

If you mean the change to another file, is just for semantics (it is a constant, not a type used to generate the CRD, so lets move to constants file)

File: config/crd/v2alpha1/bases/velero.io_datauploads.yaml (hunk 2 above)
Comment (mateusoliveira43) [reply config/crd/v2alpha1/bases/velero.io_datauploads.yaml:119]: The code implicitly tell us that. If the intent is to let an open string field, would remove this type from apis file

Comment (kaovilai) [reply config/crd/v2alpha1/bases/velero.io_datauploads.yaml:119]: I believe datauploads CRD is meant for any data movers. CSI being the first one implemented by velero. Others could come in with Database snapshot uploader etc..

Comment (reasonerjt) [reply config/crd/v2alpha1/bases/velero.io_datauploads.yaml:119]: I don't think we wanna introduce frequent changes to APIs.

Comment (shubham-pampattiwar) [reply config/crd/v2alpha1/bases/velero.io_datauploads.yaml:119]: +1 @kaovilai

Comment (kaovilai) [reply config/crd/v2alpha1/bases/velero.io_datauploads.yaml:119]: This enum needs to be removed IMO. It's an open value per design.. or else comment in the design that future datauploaders need to add values to this enum

Comment (mateusoliveira43) [reply config/crd/v2alpha1/bases/velero.io_datauploads.yaml:119]: Can you point to the design?

I found this, which implies it is a enum https://github.com/vmware-tanzu/velero/blob/main/design/volume-snapshot-data-movement/volume-snapshot-data-movement.md?plain=1#L162

Comment (kaovilai) [reply config/crd/v2alpha1/bases/velero.io_datauploads.yaml:119]: >  SnapshotType is the type of the snapshot to be backed up.

https://github.com/vmware-tanzu/velero/blob/e65ef2894868242d51a92e1fba26423adb6ca49f/design/volume-snapshot-data-movement/volume-snapshot-data-movement.md?plain=1#L272-L275C25

Plan is to open up this to more values in the future.

**Comment by mateusoliveira43** [comment]
As discussed yesterday with @kaovilai @sseago and @shawn-hurley , I will close this PR as we do not want to change an API field without updating its version, so users can have the best experience possible while using the product

Will open follow up PR just to add comments on Go code, on why those fields do not use enum validation
